### PR TITLE
Add option to download selected assets zip in list view

### LINKED ISSUE #2747: Not able to download selected assets
# Improvement Request
Hello ,
There is no option to download selected assets.There is an option to download assets as zip but if there is thousand assets in that folder and i want to download selected ones may be 50 or less , I have to download those manually.

assets download

### PR DESCRIPTION
**Pull request by dvesh3**
resolves #2747 

## Changes in this pull request  
Added option in Asset list view to download selected assets from the list
download-zip

### CODE REVIEW

File: web/pimcore/static6/js/pimcore/asset/listfolder.js [hunk 1]
```diff
@@ -250,7 +256,71 @@ pimcore.asset.listfolder = Class.create({
...
+
+    downloadSelectedZip: function () {
+
+        //create the ids array which contains chosen rows to export
+        ids = [];
+        var selectedRows = this.grid.getSelectionModel().getSelection();
+        for (var i = 0; i < selectedRows.length; i++) {
+            ids.push(selectedRows[i].data.id);
+        }
+        console.log(ids);
```
Comment (fashxp) [inline web/pimcore/static6/js/pimcore/asset/listfolder.js]: please remove the console logs :-)

File: web/pimcore/static6/js/pimcore/asset/listfolder.js [hunk 2]
```diff
@@ -250,7 +256,71 @@ pimcore.asset.listfolder = Class.create({
...
+    downloadSelectedZip: function () {
+
+        //create the ids array which contains chosen rows to export
+        ids = [];
+        var selectedRows = this.grid.getSelectionModel().getSelection();
+        for (var i = 0; i < selectedRows.length; i++) {
+            ids.push(selectedRows[i].data.id);
+        }
+        
+        Ext.Ajax.request({
```
Comment (brusch) [inline web/pimcore/static6/js/pimcore/asset/listfolder.js]: This code is 99,9% identical with https://github.com/pimcore/pimcore/blob/aea60550fae338e73bdb4435a0cc0b7ae370570e/web/pimcore/static6/js/pimcore/asset/folder.js#L316
We should reuse the existing code, eg. by passing the params as an argument of `pimcore.asset.folder.downloadZip`.

File: web/pimcore/static6/js/pimcore/asset/listfolder.js [hunk 3]
```diff
@@ -148,7 +148,13 @@ pimcore.asset.listfolder = Class.create({
             tbar: [
                 "->"
                 ,this.checkboxOnlyDirectChildren
-                ]
+                , "-"
+                ,{
+                    text: t("download_as_zip"),
```
Comment (brusch) [inline web/pimcore/static6/js/pimcore/asset/listfolder.js]: We should use `Download selected items as ZIP` as the label, to differentiate from the `Download as ZIP` button in the toolbar.

File: web/pimcore/static6/js/pimcore/asset/listfolder.js [hunk 4]
```diff
@@ -148,7 +148,13 @@ pimcore.asset.listfolder = Class.create({
...
+                    iconCls: "pimcore_icon_export",
```
Comment (brusch) [inline web/pimcore/static6/js/pimcore/asset/listfolder.js]: The class `pimcore_icon_zip pimcore_icon_overlay_download` would be more consistent I'd say.

File: web/pimcore/static6/js/pimcore/asset/listfolder.js (hunk 2 above)
Comment (dvesh3) [reply web/pimcore/static6/js/pimcore/asset/listfolder.js]: Done! :)

**Comment by brusch** [comment]
jfyi: I've added some smaller improvements with https://github.com/pimcore/pimcore/pull/2830/commits/2f5b50b1d1d89efb7dbe68fbfd389d95d4423768
